cirq/sim/density_matrix_simulator.py

```diff
@@ -122,25 +122,28 @@ def __init__(self,
                  *,
                  dtype: Type[np.number] = np.complex64,
                  noise: 'cirq.NOISE_MODEL_LIKE' = None,
-                 seed: Optional[int] = None):
+                 seed: Optional[Union[int, np.random.RandomState]] = None):
         """Density matrix simulator.
 
          Args:
             dtype: The `numpy.dtype` used by the simulation. One of
                 `numpy.complex64` or `numpy.complex128`
             noise: A noise model to apply while simulating.
-            seed: The random seed to use for this simulator. Sets numpy's
-                random seed. Setting numpy's seed different in between
-                use of this class will lead to non-seeded behavior.
+            seed: The random seed to use for this simulator.
         """
         if dtype not in {np.complex64, np.complex128}:
             raise ValueError(
                 'dtype must be complex64 or complex128, was {}'.format(dtype))
 
         self._dtype = dtype
         self.noise = devices.NoiseModel.from_noise_model_like(noise)
-        if seed:
-            np.random.seed(seed)
+
+        if seed is None:
```

Comment:
This block of code is repeated in a bunch of places, with various different cases for None.  For None I'd always set it to np.random.  I'd then either stick it into a util class somewhere, or alternatively only make it as Optional in the main entry points where it is used and otherwise set it as a default to seed=np.random.

Reply:
For further color, networkx has a decorator https://github.com/networkx/networkx/blob/master/networkx/utils/decorators.py#L354 that they apply to all their functions that need randomness

They also have nice documentation https://networkx.github.io/documentation/stable/reference/randomness.html